Monika After Story/game/script-topics.rpy

```diff
@@ -12905,3 +12905,205 @@ label monika_enjoyingspring:
+    if len(derandomlist) > 1:
+        $ renpy.say(m,"Which topic are you okay with talking about again?", interact=False )
+    else:
+        $ renpy.say(m,"If you're sure it's alright to talk about this again, just click the topic.", interact=False )
+
+    call screen mas_gen_scrollable_menu(derandomlist,(evhand.UNSE_X, evhand.UNSE_Y, evhand.UNSE_W, 500), evhand.UNSE_XALIGN, return_prompt_back)
+    show monika at t11
+
+    $ topic_choice = _return
+
+    if not _return:
+        m 1eua "Okay, [player]."
+
+    else:
+        $ mas_showEVL(topic_choice,"EVE",_random=True)
+        $ del persistent._mas_player_derandomed[topic_choice]
+        m 1eua "Okay, [player]..."
+
+        if len(persistent._mas_player_derandomed) > 0:
+            m 1eka "Are there any other topics you are okay with talking about again?{nw}"
+            $ _history_list.pop()
+            menu:
+                m "Are there any other topics you are okay with talking about again?{fast}"
+                "Yes.":
+                    jump mas_topic_rerandom
+                "No.":
+                    m 3eua "Okay."
+
+        else:
+            m 3hua "All done!"
+            $ mas_hideEVL("mas_topic_rerandom","EVE",lock=True)
+
+    # make sure if we are rerandoming any seasonal specific topics, stuff that's supposed
+    # to be derandomed out of season is still derandomed
+    $ persistent._mas_current_season = store.mas_seasons._seasonalCatchup(persistent._mas_current_season)
+    return
+
+init 5 python:
+    addEvent(Event(persistent.event_database,eventlabel="mas_topic_unbookmark",prompt="I'd like to remove a bookmark.",unlocked=False,rules={"no unlock":None}))
+
+label mas_topic_unbookmark:
+    python:
+        bookmarkslist = [
+            (renpy.substitute(ev.prompt), ev_label, False, False)
+            for ev_label, ev in persistent._mas_player_bookmarked.iteritems()
+            if ev.unlocked and ev.checkAffection(mas_curr_affection)
+        ]
+
+        bookmarkslist.sort()
+        return_prompt_back = ("Nevermind.", False, False, False, 20)
+
+    show monika 1eua at t21
+    if len(bookmarkslist) > 1:
+        $ renpy.say(m,"Which bookmark do you want to remove?", interact=False )
+    else:
+        $ renpy.say(m,"Please click the bookmark if you're sure you want to remove it.", interact=False )
+        
+    call screen mas_gen_scrollable_menu(bookmarkslist,(evhand.UNSE_X, evhand.UNSE_Y, evhand.UNSE_W, 500), evhand.UNSE_XALIGN, return_prompt_back)
+
+    $ topic_choice = _return
+
+    if not topic_choice:
+        m 1eua "Okay, [player]."
+        $ pushEvent('mas_bookmarks',True)
+
+    else:
+        show monika at t11
+        $ del persistent._mas_player_bookmarked[topic_choice]
+        # since it's possible to have bookmarked items that are now locked and thus not shown here
+        # we will delete an event from the list so we can check the lenth of currently available events
+        $ del bookmarkslist[-1]
```

Comment:
can you explain in more detail why this is neccessary?

Reply:
not all events in the dict are shown here (since we exclude locked events as well as events that are unavailable due to affection, so it's possible for the length of the dict to be greater than zero, but not the list. if we just check dict length, she could ask if there's anything else you want to remove when there's nothing in the list to show.

Reply:
but why are you just directly removing the last element of the list? is that always the selected topic?

Reply:
because it doesn't matter what we remove, the list gets repopulated again before you can select again. it's solely to check length

Reply:
then just have the if statement check `len(bookmarkslist)-1`

Reply:
or do `if len(bookmarkslist) > 1`